Doc/tools/extensions/pyspecific.py

```diff
@@ -31,7 +31,16 @@
 from sphinx.util.nodes import split_explicit_title
 from sphinx.writers.text import TextWriter, TextTranslator
 from sphinx.writers.latex import LaTeXTranslator
-from sphinx.domains.python import PyModulelevel, PyClassmember
+
+try:
+    from sphinx.domains.python import PyMethod
+except ImportError:
+    from sphinx.domains.python import PyClassmember as PyMethod
+
+try:
+    from sphinx.domains.python import PyFunction
+except ImportError:
+    from sphinx.domains.python import PyModulelevel as PyFunction
```

Comment:
Would it make sense to merge the two try/except? I expect that PyMethod and PyFunction were added to the same Sphinx version.